src/data/complex/is_R_or_C.lean

```diff
@@ -197,8 +201,8 @@ def norm_sq (z : K) : ℝ := re z * re z + im z * im z
 lemma norm_sq_eq_def {z : K} : ∥z∥^2 = (re z) * (re z) + (im z) * (im z) := norm_sq_eq_def_ax z
 lemma norm_sq_eq_def' (z : K) : norm_sq z = ∥z∥^2 := by rw [norm_sq_eq_def, norm_sq]
 
-@[simp] lemma norm_sq_of_real (r : ℝ) : ∥𝓚 r∥^2 = r * r :=
-by simp [norm_sq_eq_def]
+lemma norm_sq_of_real (r : ℝ) : ∥𝓚 r∥^2 = r * r :=
+by simp [pow_two, real.norm_eq_abs, abs_mul_abs_self]
```

Comment:
This one can't be a `simp` lemma any more, since its LHS simplifies to `∥r∥^2` (at which point it's just a lemma about `ℝ` and not `K`).

Discussion about adding this as a simp lemma for `ℝ` is out of scope for this PR, but is discussed here: https://leanprover.zulipchat.com/#narrow/stream/217875-Is-there.20code.20for.20X.3F/topic/.E2.88.80.20(r.20.3A.20.E2.84.9D).2C.20.E2.88.A5r.E2.88.A5.5E2.20.3D.20r.20*.20r